File: go/lib_test.go
package main

import (
	"testing"

	"github.com/n0-computer/iroh-ffi/iroh"
	"github.com/stretchr/testify/assert"
)

/// TestPathToKeyRoundtrip
func TestPathToKeyRoundtrip(t *testing.T) {
	path := "/foo/bar"
	key := []byte("/foo/bar\x00")

	// Test without prefix and root
	gotKey, err := iroh.PathToKey(path, nil, nil)
	assert.Nil(t, err)

	assert.Equal(t, key, gotKey)

	gotPath, err := iroh.KeyToPath(gotKey, nil, nil)
	assert.Nil(t, err)

	assert.Equal(t, path, gotPath)

	// Test with prefix
	prefix := "prefix:"
	key = []byte("prefix:/foo/bar\x00")

	gotKey, err = iroh.PathToKey(path, &prefix, nil)
	assert.Nil(t, err)

	assert.Equal(t, key, gotKey)

	gotPath, err = iroh.KeyToPath(gotKey, &prefix, nil)
	assert.Nil(t, err)

	assert.Equal(t, path, gotPath)

	// Test with root
	root := "/foo"
	key = []byte("prefix:bar\x00")

	gotKey, err = iroh.PathToKey(path, &prefix, &root)
	assert.Nil(t, err)

	assert.Equal(t, key, gotKey)

	gotPath, err = iroh.KeyToPath(gotKey, &prefix, &root)
	assert.Nil(t, err)

	assert.Equal(t, path, gotPath)
}

File: python/lib_test.py
# tests that correspond to the `src/lib.rs` rust api functions
from iroh import path_to_key, key_to_path

def test_path_to_key_roundtrip():
    path = "/foo/bar"
    key = b'/foo/bar\0'
    #
    got_key = path_to_key(path, None, None)
    assert key == got_key
    #
    got_path = key_to_path(got_key, None, None)
    assert path == got_path
    #
    # including prefix
    prefix = "prefix:"
    key = b'prefix:/foo/bar\0'
    #
    got_key = path_to_key(path, prefix, None)
    assert key == got_key
    #
    got_path = key_to_path(got_key, prefix, None)
    assert path == got_path
    #
    # including root
    root = "/foo"
    key = b'prefix:bar\0'
    #
    got_key = path_to_key(path, prefix, root)
    assert key == got_key
    #
    got_path = key_to_path(got_key, prefix, root)
    assert path == got_path
